tests/unit/test_solvers/test_solution.py

```diff
@@ -346,6 +346,23 @@ def test_solution_evals_with_inputs(self):
         time = sim.solution["Time [h]"](sim.solution.t)
         self.assertEqual(len(time), 10)
 
+    def test_condition_number(self):
```

Comment:
This is not a great test, since it just copies the code in the function. If the code gives the wrong result then the test won't catch it. It's better to test that the code outputs the expected result in some specific cases. These can be very simple cases, e.g. `dx/dt = 1` and `dx/dt = x`, in which case you can calculate the jacobian (and hence the condition number) explicitly

Reply:
Another test could be to check that the condition number gets larger as the problem gets stiffer